python/letsql/common/utils/caching_utils.py

```diff
@@ -0,0 +1,80 @@
+import dask.base
+
+from ibis import BaseBackend
+from letsql.expr.relations import CachedNode, RemoteTable, Read
+from ibis.expr import operations as ops
+
+
+def transform_cached_node(op, _, **kwargs):
+    op = op.__recreate__(kwargs)
+
+    if isinstance(op, CachedNode):
+        uncached, storage = op.parent, op.storage
+
+        first, _ = find_backend(uncached.op())
+        other = storage.source
+
+        if first is not other:
+            name = dask.base.tokenize(
+                {
+                    "schema": op.schema,
+                    "expr": uncached.unbind(),
+                    "source": first.name,
+                    "sink": other.name,
+                }
+            )
+
+            parent = RemoteTable.from_expr(other, uncached, name=name).to_expr()
+            op = CachedNode(
+                schema=op.schema,
+                parent=parent,
+                source=storage.source,
+                storage=storage,
+            )
+
+    return op
+
+
+def find_backend(op: ops.Node) -> tuple[BaseBackend, bool]:
+    backends = set()
+    has_unbound = False
+    node_types = (
+        ops.UnboundTable,
+        ops.DatabaseTable,
+        ops.SQLQueryResult,
+        CachedNode,
+        Read,
+    )
+    for table in op.find(node_types):
+        if isinstance(table, ops.UnboundTable):
+            has_unbound = True
+        else:
+            backends.add(table.source)
+
+    return (
+        backends.pop(),
+        has_unbound,
+    )  # TODO what happens if it has more than one backend
+
+
+def exists(node: CachedNode):
```

Comment:
please look at `LETSQLAccessor`'s `exists` implementation
if there's duplication between `caching_utils` and `LETSQLAccessor`, let's have `LETSQLAccessor` use `caching_utils`

Reply:
This exists is a leftover, is actually not used